### ISSUE TITLE
Known issue: REST API should be opt-in

### ISSUE THREAD

**Issue by TheTinkerDad** [TheTinkerDad/sensible#13]
Right now sensible starts a minimal HTTP server on port 8090 which listens to some REST API endpoints not even present in the documentation. They are very basic, e.g. shutting down sensible, pausing and resuming MQTT message processing and hold a minimal risk especially on a local network, but still, most users probably don't need this.

The obvious solution is to make this configurable in the settings.yaml file with the config value turned off by default.

To sum things up:

- the REST API should be documented
- it must be opt-in
- port must be configurable
- some security must be added at least, like a token generated at startup which needs to be present in the URL

**Comment by TheTinkerDad** [comment]
Fixed/implemented all the items from the above list on the "latest" branch.
